### ISSUE TITLE
Mate Edges and Flush Surfaces Tools Doesn't Save The Changes

### ISSUE THREAD

**Issue by SwizzleMaster** [oddtopus/dodo#18]
**Background Info:** I'm using the latest freecad binary available for windows and the flatpak version on linux with the same result. I also reinstalled Dodo first to confirm if it's actually a bug.

```
OS: KDE Flatpak runtime (GNOME/gnome)
Word size of FreeCAD: 64-bit
Version: 0.20.1.29410 (Git)
Build type: Release
Branch: (HEAD detached at 0.20.1)
Hash: f5d13554ecc7a456fb6e970568ae5c74ba727563
Python 3.9.9, Qt 5.15.5, Coin 4.0.0, Vtk 8.2.0, OCC 7.6.2
Locale: English/United States (en_US)
Installed mods: 
  * FeedsAndSpeeds 0.4.0
  * FEM_FrontISTR 0.1.0
  * LCInterlocking
  * MeshRemodel 1.8919.0
  * POV-Ray-Rendering
  * Pyramids-and-Polyhedrons
  * Ship 2022.4.11
  * parts_library
  * dodo
```

**The problem:**
I create a frame structure using the frame branch manager, trim the members to suit. If I need to mate some surfaces or lines, I exit the frame branch manager and use the "Mate the Edges" or "Flush the Surfaces" tool. It appears to work and the lines or surface are made flush. However the blue tick icon appears on the modified frame member indicating that a recompute is required. Next time the document is recomputed the changes are undone. the moved frames revert to their original position. Apparently whatever the dodo tool does for those operations doesn't get saved to the open document and therefore gets lost in the recompute.

**Comment by balrobs** [comment]
Same problem here:

```
OS: Windows 10 Version 2009
Word size of FreeCAD: 64-bit
Version: 0.20.29177 (Git)
Build type: Release
Branch: releases/FreeCAD-0-20
Hash: 68e337670e227889217652ddac593c93b5e8dc94
Python 3.8.10, Qt 5.15.2, Coin 4.0.1, Vtk 8.2.0, OCC 7.6.2
```

**Comment by shawnhaigh** [comment]
Yes same, currently as a workaround, I have to reduce the dimensions of the base wire or solid relative to the size of the frame material to get the desired outer dimension of my frame every time I use the Frame Branch Manger. This quickly becomes unusable with additional complexity.

I would be great to have tools like "align to face", or "mate to edge" inside of Frame Branch Manager? Similar to how the trim/expand function is implemented, it would be a game changer!

Thanks

[code]
OS: Gentoo Linux (Unity/sway)
Word size of FreeCAD: 64-bit
Version: 0.21.1.33694 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 0.21.1)
Hash: f6708547a9bb3f71a4aaade12109f511a72c207c
Python 3.10.12, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.5, OCC 7.6.3
Locale: English/United States (en_US)
Installed mods: 
  * OpenDark 2023.10.30
  * dodo 1.0.0
[/code]
